Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -222,9 +222,9 @@ void _delayedEndFigureCall() {
     // closed.
     bool m_hasGeometryStarted;
     // True if this geometry can end its own figures.
-    bool m_allowsFigureCalls;
+    int m_allowsFigureCalls;
```

Comment:
Exactly what Dustin said. There are times in the code where allowing figure calls will call into other functions which will then disable it. Having a stack of allow/disallow let's the top level call be the only one to disable this permission.